### PR TITLE
[GH-28399] Errcheck jobtest

### PR DESCRIPTION
**Pull request by NilsArnlund**
#### Summary
This PR solves issue https://github.com/mattermost/mattermost/issues/28399. This is done by capturing the error and handling it appropriately.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28399

#### Release Note

```release-note

```

### CODE REVIEW

File: server/channels/api4/job_test.go [hunk 1]
```diff
@@ -36,7 +36,11 @@ func TestCreateJob(t *testing.T) {
 	t.Run("valid job as user with permissions", func(t *testing.T) {
 		received, _, err := th.SystemAdminClient.CreateJob(context.Background(), job)
 		require.NoError(t, err)
-		defer th.App.Srv().Store().Job().Delete(received.Id)
+		defer func() {
+			if result, appErr := th.App.Srv().Store().Job().Delete(received.Id); appErr != nil {
+				t.Logf("Failed to delete job (result: %v): %v", result, appErr)
+			}
```
Comment (hanzei) [inline server/channels/api4/job_test.go:42]: Let's use `require.Nil` to be consistent with the rest of the code base.

File: server/channels/api4/job_test.go [hunk 2]
```diff
@@ -261,20 +279,26 @@ func TestDownloadJob(t *testing.T) {
...
 
 	// Normal user cannot download the results of these job (not the right permission)
 	_, resp, err = th.Client.DownloadJob(context.Background(), job.Id)
 	require.Error(t, err)
 	CheckForbiddenStatus(t, resp)
-	th.SystemManagerClient.DownloadJob(context.Background(), job.Id)
-	// System manager with default permissions cannot download the results of these job (Doesn't have correct permissions)
+
 	_, resp, err = th.SystemManagerClient.DownloadJob(context.Background(), job.Id)
+	// System manager with default permissions cannot download the results of these job (Doesn't have correct permissions)
```
Comment (hanzei) [inline server/channels/api4/job_test.go:301]: Unrelated change?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution! Please take a took at my comments below.

File: server/channels/api4/job_test.go (hunk 1 above)
Comment (NilsArnlund) [reply server/channels/api4/job_test.go]: Yes, should be fixed

File: server/channels/api4/job_test.go (hunk 2 above)
Comment (NilsArnlund) [reply server/channels/api4/job_test.go]: Yes I messed upp a bit, now added the correct handling and removed the unwanted changes

**Comment by hanzei** [comment]
@NilsArnlund There is still one issue that the linter reports. Can you please take a look? Thanks!
`Error: channels/api4/job_test.go:41:4: calling require.Nil on error, please use require.NoError instead`

File: server/channels/api4/job_test.go [hunk 3]
```diff
@@ -36,7 +36,10 @@ func TestCreateJob(t *testing.T) {
 	t.Run("valid job as user with permissions", func(t *testing.T) {
 		received, _, err := th.SystemAdminClient.CreateJob(context.Background(), job)
 		require.NoError(t, err)
-		defer th.App.Srv().Store().Job().Delete(received.Id)
+		defer func() {
+			result, appErr := th.App.Srv().Store().Job().Delete(received.Id)
+			require.NoError(t, appErr, "Failed to delete job (result: %v): %v", result, appErr)
```
Comment (hanzei) [inline server/channels/api4/job_test.go:41]: `require.NoError` does not accept a format string. You need to use `require.NoErrorF` instead

File: server/channels/api4/job_test.go [hunk 4]
```diff
@@ -58,7 +61,11 @@ func TestGetJob(t *testing.T) {
 	_, err := th.App.Srv().Store().Job().Save(job)
 	require.NoError(t, err)
 
-	defer th.App.Srv().Store().Job().Delete(job.Id)
+	defer func() {
+		if result, appErr := th.App.Srv().Store().Job().Delete(job.Id); appErr != nil {
```
Comment (hanzei) [inline server/channels/api4/job_test.go:65]: Can you please remove the `if` check here and in the rest of the file and use `require.Nil` instead?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: server/channels/api4/job_test.go (hunk 3 above)
Comment (NilsArnlund) [reply server/channels/api4/job_test.go:41]: Ok, my bad. Should be fixed now

Comment (hanzei) [reply server/channels/api4/job_test.go:41]: Do you need to push the changes?

File: server/channels/api4/job_test.go (hunk 4 above)
Comment (NilsArnlund) [reply server/channels/api4/job_test.go]: I did mean this for the comment below. I did not get the `require.NoErrorF` to work. I got the following error: `channels/api4/job_test.go:41:12: undefined: require.NoErrorF (typecheck)`. Changed the line to: `require.NoError(t, appErr, "Failed to delete job (result: %v)", result)`. Not sure if this works though. It passed all test using `NoError` so i don't know

Comment (NilsArnlund) [reply server/channels/api4/job_test.go]: Error: channels/api4/job_test.go:66:3: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:125:4: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:208:4: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:280:3: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:341:3: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:378:4: calling require.Nil on error, please use require.NoError instead
Error: channels/api4/job_test.go:433:4: calling require.Nil on error, please use require.NoError instead

Comment (hanzei) [reply server/channels/api4/job_test.go]: My bad. `App.Srv().Store().Job().Delete` does return an `error`. So, `require.NoError` is indeed the right method to use.

File: server/channels/api4/job_test.go (hunk 3 above)
Comment (hanzei) [reply server/channels/api4/job_test.go]: Sorry, the function name is `require.NoErrorf`.

**Comment by NilsArnlund** [comment]
Sorry for the delay, I have been sick for the last couple of days but now the problems are fixed @hanzei

File: server/channels/api4/job_test.go [hunk 5]
```diff
@@ -261,18 +275,25 @@ func TestDownloadJob(t *testing.T) {
 	// as a system admin, we should get a not found status.
 	_, err = th.App.Srv().Store().Job().Save(job)
 	require.NoError(t, err)
-	defer th.App.Srv().Store().Job().Delete(job.Id)
+	defer func() {
+		_, delErr := th.App.Srv().Store().Job().Delete(job.Id)
+		require.Nil(t, delErr, "Failed to delete job %s", job.Id)
```
Comment (hanzei) [inline server/channels/api4/job_test.go:280]: There is still an issue with this line
```
Error: channels/api4/job_test.go:280:3: calling require.Nil on error, please use require.NoError instead

```

File: server/channels/api4/job_test.go [hunk 6]
```diff
@@ -261,18 +275,25 @@ func TestDownloadJob(t *testing.T) {
 	// as a system admin, we should get a not found status.
 	_, err = th.App.Srv().Store().Job().Save(job)
 	require.NoError(t, err)
-	defer th.App.Srv().Store().Job().Delete(job.Id)
+	defer func() {
+		_, delErr := th.App.Srv().Store().Job().Delete(job.Id)
+		require.NoError(t, delErr, "Failed to delete job %s", job.Id)
+	}()
```
Comment (hanzei) [inline server/channels/api4/job_test.go:281]: The file is not `go fmt`ed
```suggestion
	}()
```

File: server/channels/api4/job_test.go [hunk 7]
```diff
@@ -400,7 +428,10 @@ func TestUpdateJobStatus(t *testing.T) {
 	for _, job := range jobs {
 		_, err := th.App.Srv().Store().Job().Save(job)
 		require.NoError(t, err)
-		defer th.App.Srv().Store().Job().Delete(job.Id)
+		defer func(jobId string) {
+			_, delErr := th.App.Srv().Store().Job().Delete(jobId)
+			require.NoError(t, delErr, "Failed to delete job %s", jobId)
```
Comment (hanzei) [inline server/channels/api4/job_test.go:433]: Nit:
```suggestion
		defer func(jobID string) {
			_, delErr := th.App.Srv().Store().Job().Delete(jobID)
			require.NoError(t, delErr, "Failed to delete job %s", jobID)
```

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by calebroseland — APPROVED** [review]
